test_autofit/unit/graphical/test_transform.py

```diff
@@ -0,0 +1,296 @@
+
+import numpy as np
+
+import pytest
+import numpy as np
+from scipy import stats, linalg, optimize
+
+import autofit.graphical as graph
+import autofit.graphical.factor_graphs.transform as transform
+
+def test_CholeskyTransform():
+    d = 10
+    A = stats.wishart(d, np.eye(d)).rvs()
+
+    cho_factor = transform.CholeskyTransform(linalg.cho_factor(A))
+
+    U = np.triu(cho_factor.U)
+    iU = np.linalg.inv(U)
+
+    b = np.random.rand(d)
+    assert np.allclose(cho_factor * b, U @ b)
+    assert np.allclose(b * cho_factor, b @ U)
+    assert np.allclose(cho_factor.ldiv(b), iU @ b)
+    assert np.allclose(b / cho_factor, b @ iU)
+    
+    b = np.random.rand(d, d)
+    assert np.allclose(cho_factor * b, U @ b)
+    assert np.allclose(b * cho_factor, b @ U)
+    assert np.allclose(cho_factor.ldiv(b), iU @ b)
+    assert np.allclose(b / cho_factor, b @ iU)
+    
+    b = np.random.rand(d, d + 1)
+    assert np.allclose(cho_factor * b, U @ b)
+    assert np.allclose(cho_factor.ldiv(b), iU @ b)
+
+    b = np.random.rand(d + 1, d)
+    assert np.allclose(b * cho_factor, b @ U)
+    assert np.allclose(b / cho_factor, b @ iU)
+
+def test_DiagonalTransform():
```

Comment:
function names should be lower case.

You can group tests into test cases which are classes. In this case you could then break this test into several test methods within the test case